UnityProject/Assets/Scripts/Mapping/Move/LocationMarker.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Linq;
+using AddressableReferences;
+using Core.Physics;
+using Logs;
+using Managers;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+using UnityEngine;
+
+namespace Mapping.Move
+{
+	[RequireComponent(typeof(RegisterItem))]
+	public class LocationMarker : MonoBehaviour, IServerSpawn, IServerDespawn
+	{
+		public int LocationID = 0;
+		public RegisterItem Register;
+
+		private Vector3Int badPosition = new Vector3Int(0, 0, -100);
+
+		private void Awake()
+		{
+			Register ??= GetComponent<RegisterItem>();
+		}
+
+		public void OnSpawnServer(SpawnInfo info)
+		{
+			GameManager.Instance.AddLocationMarker(this);
+		}
+
+		public void OnDespawnServer(DespawnInfo info)
+		{
+			GameManager.Instance.RemoveLocationMarker(this);
```

Comment:
why do you need a static list of them?

Reply:
It's not a static list.

Reply:
have you tried reading the code

Reply:
the code is self-documenting, you should be able to tell why it's put like this

Reply:
so, Explain the concept to me?